packages/cube/lib/cube/buildCubeShape/Constraints.js

```diff
@@ -0,0 +1,210 @@
+export class CompositeConstraintBuilder {
+  constructor(...builders) {
+    this.builders = builders
+  }
+
+  add(object) {
+    this.builders.forEach(builder => builder.add(object))
+  }
+
+  build(ptr) {
+    this.builders.forEach(builder => builder.build(ptr))
+  }
+}
+
+export class NodeKindConstraintBuilder {
+  #termTypes = new Set()
```

Comment:
`#` prefix is a convention for private fields?

Reply:
yes, according to [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_properties)

Reply:
Oh, that's not just syntactic sugar. I spend too much in TS only :)